public/app/plugins/datasource/cloudwatch/components/PanelQueryEditor.tsx

```diff
@@ -25,9 +25,22 @@ export class PanelQueryEditor extends PureComponent<Props> {
           <Segment
             value={apiModes[apiMode]}
             options={Object.values(apiModes)}
-            onChange={({ value }) =>
-              this.props.onChange({ ...query, apiMode: (value as 'Metrics' | 'Logs') ?? 'Metrics' })
-            }
+            onChange={({ value }) => {
+              const newMode = (value as 'Metrics' | 'Logs') ?? 'Metrics';
+              if (newMode === 'Metrics') {
+                this.props.onChange({
+                  ...(query as CloudWatchMetricsQuery),
+                  apiMode: newMode,
+                  queryMode: newMode,
```

Comment:
I think this may not be correct. In explore we have more combinations. Ie we can have apiMode = logs but queryMode = metrics. So this should probably not be in sync but as 2 separate things. One pushed from query editor second from explore.

Reply:
`queryMode` is just supposed to exist in service of [discriminated unions](https://www.typescriptlang.org/docs/handbook/advanced-types.html#discriminated-unions), though in retrospect it may not be the best name.
In general I think I need to take a look at `CloudWatchMetricsQuery` and `CloudWatchLogsQuery` as I think the way they're handled right now isn't great and a bit confusing

Reply:
Ah ok right we actually do not care about the explore mode in the datasource. In that case I assume we may actually remove the queryMode? I think it was present only in Explore (and I introduced apiMode by copying code from dashboards editor) so I think it should be safe to remove.